mlflow/langchain/__init__.py

```diff
@@ -460,16 +460,34 @@ def predict(  # pylint: disable=unused-argument
         data: Union[pd.DataFrame, List[Union[str, Dict[str, Any]]], Any],
         params: Optional[Dict[str, Any]] = None,  # pylint: disable=unused-argument
     ) -> List[str]:
-        """
-        :param data: Model input data.
-        :param params: Additional parameters to pass to the model for inference.
+        messages = self._prepare_messages(data)
+        from mlflow.langchain.api_request_parallel_processor import process_api_requests
 
-                       .. Note:: Experimental: This parameter may change or be removed in a future
-                                               release without warning.
+        return_first_element = isinstance(self.lc_model, lc_runnables_types()) and not isinstance(
+            data, pd.DataFrame
+        )
+        results = process_api_requests(lc_model=self.lc_model, requests=messages)
+        return results[0] if return_first_element else results
 
-        :return: Model predictions.
-        """
+    @experimental
+    def _predict_with_callbacks(  # pylint: disable=unused-argument
+        self,
+        data: Union[pd.DataFrame, List[Union[str, Dict[str, Any]]], Any],
+        params: Optional[Dict[str, Any]] = None,  # pylint: disable=unused-argument
+        callback_handlers=None,
+    ) -> List[str]:
```

Comment:
Can we add a brief pydoc string here just for future reference purposes to explain why we have this private method here?

Reply:
I accidentally removed these during the refactor. Added back.